lib/Gateway.js

```diff
@@ -771,43 +771,65 @@ function setDiscoveryValue (payload, prop, node) {
-    )
-    cfg.discovery_payload.rgb_command_topic =
-      cfg.discovery_payload.rgb_state_topic + '/set'
-    cfg.values = [rgbID]
-
-    const whiteID = `51-${valueId.endpoint || 0}-currentColor-0`
-    const whiteValue = node.values[whiteID]
-
-    if (whiteValue) {
-      const whiteIDWrite = `51-${valueId.endpoint || 0}-targetColor-0`
-      const whiteValueWrite = node.values[whiteIDWrite]
-      cfg.discovery_payload.white_value_state_topic = this.mqtt.getTopic(
-        this.valueTopic(node, whiteValue)
-      )
-      cfg.discovery_payload.white_value_scale = whiteValue.max
-      cfg.discovery_payload.white_value_template = '{{ value_json.value }}'
-      cfg.discovery_payload.white_value_command_topic = this.mqtt.getTopic(
-        this.valueTopic(node, whiteValueWrite),
-        true
-      )
 
-      cfg.values.push(whiteID, whiteIDWrite)
-    }
+function addRgbColorSwitch (cfg, node, valueId) {
+  cfg.discovery_payload.rgb_state_topic = this.mqtt.getTopic(
+    this.valueTopic(node, valueId)
+  )
+  cfg.discovery_payload.rgb_command_topic =
+    cfg.discovery_payload.rgb_state_topic + '/set'
+
+  // The following part of code, checks if ML or Binary works. If one exists the other
+  let brightnessValue = false
+  let switchValue = false
+  if (node.values[`38-${valueId.endpoint}-currentValue`]) {
+    brightnessValue = `38-${valueId.endpoint}-currentValue`
+  } else if (node.values[`38-${valueId.endpoint + 1}-currentValue`]) {
+    brightnessValue = `38-${valueId.endpoint + 1}-currentValue`
+  } else if (node.values[`37-${valueId.endpoint}-currentValue`]) {
+    switchValue = `37-${valueId.endpoint}-currentValue`
+  }
 
-    return cfg
+  /* Find the control switch of the device Brightness or Binary
+     If multilevel is not there use binary
+     Some devices use also endpoint + 1 as on/off/brightness... try to guess that too!
+  */
+  let discoveredStateTopic
+  let discoveredCommandTopic
+  if (brightnessValue || switchValue) {
+    discoveredStateTopic = this.mqtt.getTopic(
+      this.valueTopic(node, node.values[brightnessValue])
+    )
+    discoveredCommandTopic =
+      this.mqtt.getTopic(
+        this.valueTopic(node, node.values[brightnessValue], true).targetTopic
+      ) + '/set'
+  }
+  if (brightnessValue) {
+    cfg.discovery_payload.brightness_state_topic = discoveredStateTopic
+    cfg.discovery_payload.brightness_command_topic = discoveredCommandTopic
+    cfg.discovery_payload.state_topic = discoveredStateTopic
+    cfg.discovery_payload.command_topic = discoveredCommandTopic
+  } else if (switchValue) {
+    cfg.discovery_payload.state_topic = discoveredStateTopic
+    cfg.discovery_payload.command_topic = discoveredCommandTopic
+
+    cfg.discovery_payload.state_value_template = '{{ value_template.json }}'
+    cfg.discovery_payload.on_command_type = 'last'
   }
 
-  return false
+  // Use color_value_* from HASS discovery as white switch
+  if (node.values['51-0-currentcolor']) {
```

Comment:
```suggestion
  if (node.values[`51-${valueId.endpoint}-currentColor`]) {
```

Reply:
more stuff need change here! i added some changed + `-0`in end